SofaKernel/modules/SofaBaseTopology/TopologyData.inl

```diff
@@ -57,7 +57,7 @@ void TopologyDataImpl <TopologyElementType, VecT>::createTopologicalEngine(sofa:
         this->m_topologicalEngine->setNamePrefix(std::string(sofa::core::topology::TopologyElementInfo<TopologyElementType>::name()) + std::string("Engine_"));
         if (this->getOwner() && dynamic_cast<sofa::core::objectmodel::BaseObject*>(this->getOwner())) dynamic_cast<sofa::core::objectmodel::BaseObject*>(this->getOwner())->addSlave(this->m_topologicalEngine.get());
         this->m_topologicalEngine->init();
-        this->linkToElementDataArray();
+        this->linkToElementDataArray((TopologyElementType*)NULL);
```

Comment:
The idea is to get rid of the need to have per-element-type child classes to TopologyDataImpl for the only purpose of overloading linkToElementDataArray(). Now instead this method takes a (null) pointer with the type of the element, such that the right one is called, based on the already-known template type. So yes it will call linkToPointDataArray, but only if TopologyElementType is indeed Point.